numpy/core/src/umath/ufunc_object.c

```diff
@@ -1880,12 +1879,15 @@ _parse_axes_arg(PyUFuncObject *ufunc, PyObject *axes, PyArrayObject **op,
                 j++;
             }
         }
-        do_remap[i] = (j < op_ndim);
+        if (j == op_ndim) {
+            remap_axis[i] = NULL;
+        }
     } /* end of for(i) loop over operands */
     return 0;
 }
 
-#define REMAP_AXIS(i, j) ((axes != NULL && do_remap[i]) ? remap_axis[i][j] : j)
+#define REMAP_AXIS(i, j) ((remap_axis != NULL && \
+                           remap_axis[i] != NULL) ? remap_axis[i][j] : j)
```

Comment:
I'm not convinced this is clearer or faster than having a 1d and 2d array

Reply:
I can undo the last commit, which changed it to a memory allocation. I mostly disliked pre-allocating quite a chunk of memory for a case that is far from common, and then having to pass on three arrays for just one goal.